gammapy/modeling/models/spatial.py

```diff
@@ -1580,6 +1584,11 @@ def __init__(
         self.meta = dict() if meta is None else meta
         if filename is not None:
             filename = str(make_path(filename))
+        if filename is None:
+            log.warning(
+                "The filename is not defined therefore the model will not be serialised correctly. "
+                'To set the filename the "model.filename" attribute can be used.'
+            )
```

Comment:
maybe a function `_check_filename(model)` could be introduced in gammapy.modeling.models.io